### PR TITLE
BE-7-add-validation

### PR DESCRIPTION
**Pull request by Randayy**
image
if user try to view/update/delete user with not it`s id he will see that 
image

### CODE REVIEW

File: app/services/user_service.py [hunk 1]
```diff
@@ -113,33 +148,9 @@ async def create_user_from_token(self, email: str) -> UserDetailSchema:
...
-        if current_date > exp_date:
-            raise HTTPException(
-                status_code=401, detail="Token has expired")
-        return old_user
+    async def check_user_permissions(self, user_id: UUID, current_user_id: UUID) -> bool:
+        if current_user_id != user_id:
+            raise HTTPException(status_code=status.HTTP_403_FORBIDDEN,
+                                detail="You dont have permission to access this user")
+        else:
+            return True
```
Comment (Artemooon) [inline app/services/user_service.py]: discussed it on the call, you do not need to return True here, because raise will throw an exception already

Comment (Randayy) [reply app/services/user_service.py]: ok I remove it

File: app/services/user_service.py [hunk 2]
```diff
@@ -17,11 +17,43 @@
...
+    try:
+        payload = jwt.decode(token.credentials, settings.auth0_secret_key,
+                             algorithms=[settings.jwt_algorithm], audience=settings.auth0_audience, issuer=settings.auth0_issuer)
+        username: str = payload.get("sub")
+        email: str = payload.get("email")
+        exp: int = payload.get("exp")
+        if username is None:
+            raise HTTPException(
+                status_code=401, detail="No Username found in token")
+        if email is None:
```
Comment (Artemooon) [inline app/services/user_service.py]: if not email, same with the username

Comment (Randayy) [reply app/services/user_service.py]: changed

File: app/services/user_service.py [hunk 3]
```diff
@@ -49,31 +84,31 @@ async def get_users_list_paginated(self, page: int, limit: int) -> List[UserDeta
         users = await self.user_repository.get_users_list_paginated(page, limit)
         return UserListSchema(users=[UserDetailSchema.from_orm(user) for user in users])
 
-    async def delete_user(self, user_id: UUID) -> None:
-        await self.user_repository.delete_user(user_id)
-        return None
+    async def delete_user(self, user_id: UUID, current_user: User) -> None:
+        result = await self.check_user_permissions(user_id, current_user.id)
+        if result:
```
Comment (Artemooon) [inline app/services/user_service.py]: you will not get result here
just do 
```
await self.check_user_permissions(user_id, current_user.id)
await self.user_repository.delete_user(user_id)
```

File: app/services/user_service.py [hunk 4]
```diff
@@ -37,9 +69,12 @@ async def create_user(self, user_data: SignUpRequestSchema) -> User:
         user = await self.user_repository.create_user(user_data_dict)
         return user
 
-    async def get_user_by_id(self, user_id: UUID) -> UserDetailSchema:
+    async def get_user_by_id(self, user_id: UUID, current_user: User) -> UserDetailSchema:
         user = await self.user_repository.get_user_by_id(user_id)
-        return user
+        current_user_id = current_user.id
+        result = await self.check_user_permissions(user_id, current_user_id)
```
Comment (Artemooon) [inline app/services/user_service.py:75]: here the same

File: app/services/user_service.py (hunk 3 above)
Comment (Randayy) [reply app/services/user_service.py]: fixed

File: app/services/user_service.py (hunk 4 above)
Comment (Randayy) [reply app/services/user_service.py:75]: fixed

File: app/schemas/user_schemas.py [hunk 5]
```diff
@@ -1,41 +1,60 @@
...
-    username: str = Field(..., min_length=6, description="Username must be at least 6 characters long")
+    username: str = Field(..., min_length=6,
+                          description="Username must be at least 6 characters long")
     email: str
     first_name: Optional[str] = None
     last_name: Optional[str] = None
 
 
+class UserBaseSchema(BaseModel):
+    username: str = Field(..., min_length=6,
```
Comment (Artemooon) [inline app/schemas/user_schemas.py]: what if user wants to update only password without username? make it optional

Comment (Randayy) [reply app/schemas/user_schemas.py]: added optional to username

**Review by Artemooon — APPROVED** [review]

**Pull request merged**